src/many/src/types/ledger.rs

```diff
@@ -426,11 +431,76 @@ impl Transaction {
                 withdrawer,
                 ..
             } => id == withdrawer || id == account,
+            TransactionInfo::MultisigSetDefaults {
+                account, submitter, ..
+            } => submitter.map(|ref e| id == e).unwrap_or(false) || id == account,
+        }
+    }
+}
+
+// This is necessary because variables must be used in repeating patterns.
+macro_rules! replace_expr {
+    ($_t:tt $sub:expr) => {
+        $sub
+    };
+}
+
+macro_rules! encode_tx_info {
+    ( $( $sname: ident { $( $idx: literal => $name: ident : $type: ty, )* }, )* ) => {
+        impl Encode for TransactionInfo {
+            fn encode<W: encode::Write>(
+                &self,
+                e: &mut Encoder<W>,
+            ) -> Result<(), encode::Error<W::Error>> {
+                match self {
+                    $(  TransactionInfo :: $sname { $( $name, )* } => {
+                            e.map( 1u64 $(+ replace_expr!($idx 1u64))* )?
```

Comment:
This doesn't seem to fit what's in the spec in PR https://github.com/many-protocol/specification/pull/35/files

`{ 0 => TransactionKind::foo, 1 => foo@param }`

Also, I think the naming should be revised; there's a `ledger-transaction-type-send` in the spec, but here it's a `TransactionKind`, `ledger-transaction-info` vs `account-multisig-transaction` which are both `TransactionInfo`, but not really.

Reply:
Let's discuss? I think you got the wrong thing or maybe outdated.